packages/offix-datastore/src/Model.ts

```diff
@@ -89,4 +133,34 @@ export class Model<T = unknown> {
                 listener(event);
             });
     }
+
+    private async subscribeForServerEvents(replicator: IReplicator) {
+        // TODO replicator.subscribe
+        // handle subscription events here
+    }
+
+    private async doDeltaSync(replicator: IReplicator, matcher: (d: T) => Predicate<T>, predicate?: Predicate<T>) {
+        // TODO limit the size of data returned
+        // TODO get lastSync for model for metadata store and pass to pullDelta
+        const data = await replicator.pullDelta(this.getStoreName(), "", predicate);
```

Comment:
[Question] How we avoid user app being bombarded by many events when multiple items were returned (this is different from subscriptions that are only per single item)

Reply:
Remember that we intend to move events.
Here, we would not use normal model crud methods(which emits events per action) but instead call storage directly